sigopt/xgboost/compute_metrics.py

```diff
@@ -1,19 +1,67 @@
 import numpy
-from sklearn.metrics import (
-  accuracy_score,
-  classification_report,
-  mean_absolute_error,
-  mean_squared_error
-)
 
 
+def compute_positives_and_negatives(y_true, y_pred, class_label):
+  y_true_equals = y_true == class_label
+  y_true_notequals = y_true != class_label
+  y_pred_equals = y_pred == class_label
+  y_pred_notequals = y_pred != class_label
+  tp = numpy.count_nonzero(numpy.logical_and(y_true_equals, y_pred_equals))
+  tn = numpy.count_nonzero(numpy.logical_and(y_true_notequals, y_pred_notequals))
+  fp = numpy.count_nonzero(numpy.logical_and(y_true_notequals, y_pred_equals))
+  fn = numpy.count_nonzero(numpy.logical_and(y_true_equals, y_pred_notequals))
+  return tp, tn, fp, fn
+
+
+def compute_accuracy(y_true, y_pred):
+  accuracy = numpy.count_nonzero(y_true == y_pred) / len(y_true)
+  return accuracy
+
+
+def compute_classification_report(y_true, y_pred):
+  classes = numpy.unique(y_true)
+  classification_report = {}
+  classification_report['weighted avg'] = {
+    'f1-score': 0,
+    'recall': 0,
+    'precision': 0
+  }
+  for class_label in classes:
+    tp, _, fp, fn = compute_positives_and_negatives(y_true, y_pred, class_label)
+    precision = tp / (tp + fp) if (tp + fp)!=0 else 0
+    recall = tp / (tp + fn) if (tp + fn)!=0 else 0
+    f1 = tp / (tp + 0.5 * (fp + fn)) if (tp + 0.5 * (fp + fn))!=0 else 0
+    support = numpy.count_nonzero(y_true == class_label)
+    classification_report[str(class_label)] = {
+      'precision': precision,
+      'recall': recall,
+      'f1-score': f1,
+      'support': support
+    }
+    classification_report['weighted avg']['precision'] += (support / len(y_pred)) * precision
+    classification_report['weighted avg']['recall'] += (support / len(y_pred)) * recall
+    classification_report['weighted avg']['f1-score'] += (support / len(y_pred)) * f1
```

Comment:
That's true, but it does seem like an reasonable way of storing these metrics... inclined to keep unless you have another suggestion

Reply:
Since we don't report/log the per-class metrics here, storing them seems unnecessary. However, I do see a case where maybe we want to return per-class metric for binary-classification problems, then this is probably fine as is.